src/org/benf/cfr/reader/bytecode/analysis/parse/expression/ArithmeticOperation.java

```diff
@@ -90,6 +92,174 @@ public Dumper dumpInner(Dumper d) {
         return d;
     }
 
+    @Override
+    public Literal getComputedLiteral(Map<LValue, Literal> display) {
+        Literal l = lhs.getComputedLiteral(display);
+        if (l == null || !(l.getValue().getValue() instanceof Number))
+            return null;
+        Literal r = rhs.getComputedLiteral(display);
+        if (r == null || !(r.getValue().getValue() instanceof Number))
+            return null;
+        RawJavaType lType = l.getInferredJavaType().getRawType();
+        RawJavaType rType = r.getInferredJavaType().getRawType();
+        Number lv = ((Number) l.value.getValue());
+        Number rv = ((Number) r.value.getValue());
+        RawJavaType widestType = lType.ordinal() > rType.ordinal() ? lType : rType;
+        switch (widestType) {
+            case BOOLEAN:
+            case BYTE:
+            case CHAR:
+            case SHORT:
+            case INT:
+                return getComputedInt(rv.intValue(), lv.intValue());
```

Comment:
I just don't think treating all things as integers is safe - 

[CastTest2.txt](https://github.com/leibnitz27/cfr/files/5549380/CastTest2.txt)

Reply:
I believe this is correct as it falls under primitive widening. The issue with the `CastTest2` seems to be improperly handling the value in the `CastExpression` computation.

Reply:
I've pushed the fix for this in `CastExpression` where `CastTest2.txt` now prints `-2` as the computed byte value.

Reply:
Marking this as resolved with the inclusion of some of the recent changes _(and paste from your example)_